src/utils/verifyStateShape.js

```diff
@@ -0,0 +1,24 @@
+/**
+ *  Shallowly compares the shape of the initial state supplied to createStore with
+ *  the shape of the state returned by the reducer. Returns a warning
+ *  if they differ.
+ *
+ * @param {any} initialState Initial state supplied to createStore.
+ *
+ * @param {any} currentState Current state of the store.
+ *
+ * @returns {string} Warning message if initial and current state do
+ * not have the same shape.
+ */
+
+import isPlainObject from './isPlainObject';
+
+export default function verifyStateShape(initialState, currentState) {
```

Comment:
Let's put it into `combineReducers.js`. I don't see the benefit of keeping it separately.